extensions/kafka/src/main/java/io/deephaven/kafka/ingest/GenericRecordChunkAdapter.java

```diff
@@ -47,63 +50,73 @@ public static GenericRecordChunkAdapter make(
-                            // micros to nanos
-                            return new GenericRecordLongFieldCopierWithMultiplier(fieldName, 1000L);
-                        } else if (LogicalTypes.timestampMillis().equals(logicalType)) {
-                            // millis to nanos
-                            return new GenericRecordLongFieldCopierWithMultiplier(fieldName, 1000_000L);
-                        }
-                        throw new IllegalArgumentException(
-                                "Can not map field with unknown logical type to DateTime: field=" + fieldName
-                                        + ", type=" + logicalType);
-
-                    } else {
+                    final String[] fieldPath = GenericRecordUtil.getFieldPath(fieldPathStr, separator);
+                    final Schema fieldSchema = GenericRecordUtil.getFieldSchema(schema, fieldPath);
+                    final LogicalType logicalType = fieldSchema.getLogicalType();
+                    if (logicalType == null) {
                         throw new IllegalArgumentException(
-                                "Can not map field not in schema to DateTime: field=" + fieldName);
+                                "Can not map field without a logical type to DateTime: field=" + fieldPathStr);
+                    }
+                    if (logicalType instanceof LogicalTypes.TimestampMillis) {
+                        return new GenericRecordLongFieldCopierWithMultiplier(fieldPathStr, separator, 1000_000L);
                     }
+                    if (logicalType instanceof LogicalTypes.TimestampMicros) {
+                        return new GenericRecordLongFieldCopierWithMultiplier(fieldPathStr, separator, 1000L);
+                    }
+                    throw new IllegalArgumentException(
+                            "Can not map field with unknown logical type to DateTime: field=" + fieldPathStr
+                                    + ", type=" + logicalType);
+
                 }
-                return new GenericRecordLongFieldCopier(fieldName);
+                return new GenericRecordLongFieldCopier(fieldPathStr, separator);
             case Float:
-                return new GenericRecordFloatFieldCopier(fieldName);
+                return new GenericRecordFloatFieldCopier(fieldPathStr, separator);
             case Double:
-                return new GenericRecordDoubleFieldCopier(fieldName);
+                return new GenericRecordDoubleFieldCopier(fieldPathStr, separator);
             case Object:
                 if (dataType == String.class) {
-                    return new GenericRecordStringFieldCopier(fieldName);
-                } else {
-                    return new GenericRecordObjectFieldCopier(fieldName);
+                    return new GenericRecordStringFieldCopier(fieldPathStr, separator);
+                }
+                final Schema.Field field = schema.getField(fieldPathStr);
```

Comment:
Do we still need this lookup or null check? I feel like this code should be more like the `DateTime` code:
```
                if (dataType == BigDecimal.class) {
                    final String[] fieldPath = GenericRecordUtil.getFieldPath(fieldPathStr, separator);
                    final Schema fieldSchema = GenericRecordUtil.getFieldSchema(schema, fieldPath);
                    final LogicalType logicalType = fieldSchema.getLogicalType();
                    if (logicalType instanceof LogicalTypes.Decimal) {
                        final LogicalTypes.Decimal decimalType = (LogicalTypes.Decimal) logicalType;
                        return new GenericRecordBigDecimalFieldCopier(
                                fieldPathStr, separator,
                                decimalType.getPrecision(), decimalType.getScale());
                    }
                    // Fall through to GenericRecordObjectFieldCopier? Or:
                    throw new IllegalArgumentException(
                            "Can not map field with unknown logical type to BigDecimal: field=" + fieldPathStr
                                    + ", type=" + logicalType);
                }
```